### ISSUE TITLE
SKU search is not returning result correctly

### ISSUE THREAD

**Issue by flexsingit** [Smile-SA/elasticsuite#1350]
### Preconditions

Magento Version : 2.3.0

ElasticSuite Version : 2.7.0

Environment : Default

Third party modules : No

### Steps to reproduce

1. we have installed and set up elasticsuite with Magento 2.3 and elastics search is working pretty good when we tested with some SKU. but there are some instances of the search in which it is not returning the result as expected. For example, when we search with the ISD-TN450 it is returning the correct results i.e 3 products but when we search with the ISD-TN45 it is not returning any result.

2. We have checked the demo at http://demo.magento-elastic-suite.io/ and search with the keyword WT09 it is returning the exact result but when we search with the keyword WT0 it is not showing any result. but the product exists and showing on the website. Please suggest if we are missing something in our configuration.
3.

### Expected result

1. When we search with the ISD-TN450 it is returning the correct results i.e 3 products but when we search with the ISD-TN45 it is not returning any result. it should return the result in both cases.

### Actual result

1. [Screenshot, logs]

**Comment by romainruaud** [comment]
Short version : It's working as intended and as provided by ElasticSuite legacy.

Actually, since every merchants have a different needs about how SKU-based searches should be handled, we cannot cover all cases on the legacy module.

@rbayet will write a small guide to explain whan can/cannot do the legacy implementation, and how it can be customized to fit any customer's needs.

That being said, for these kinds of demands, we can propose you Commercial Support to help customizing SKU-based search results on your project. 

Regards

**Comment by pmsteil** [comment]
I just found ElasticSuite today and considering it for use on our new Magento system for our client.  Looks like a great project with lots of great functionality!

I saw this open issue and the answer and I am wondering if I am missing something.

It seems to me that @flexsingit is just asking for the end user to be able to do a "fuzzy" search on the SKU.  Shouldn't that be something that is enabled by using elasticsearch/elasticsuite?

Let me know if I am missing something.  I know my customer will want this functionality also.  

Thanks!
